From 5143268c22628c5dfc620dbb746e3bab8c6ba779 Mon Sep 17 00:00:00 2001
From: Christian Stewart <christian@paral.in>
Date: Tue, 28 Jun 2022 17:23:33 -0700
Subject: [PATCH] starfive: jh7100: use distro boot logic

Remove the custom "fedora" boot env changes and use the default distro bootcmd
which scans mmc partitions for a boot.scr or extlinux.conf.

Signed-off-by: Christian Stewart <christian@paral.in>
---
 include/configs/starfive-jh7100.h | 33 +------------------------------
 1 file changed, 1 insertion(+), 32 deletions(-)

diff --git a/include/configs/starfive-jh7100.h b/include/configs/starfive-jh7100.h
index ab9a2f149f..3e55bf64a8 100644
--- a/include/configs/starfive-jh7100.h
+++ b/include/configs/starfive-jh7100.h
@@ -55,41 +55,10 @@
 
 #include <environment/distro/sf.h>
 
-#define STARLIGHT_FEDORA_BOOTENV \
-	"bootdir=/boot\0" \
-	"bootenv=uEnv.txt\0" \
-	"mmcdev=0\0" \
-	"mmcpart=3\0"
-
-#define STARLIGHT_TEST_BOOTENV \
-	"testpart=0:1\0" \
-	"testenv=u74_uEnv.txt\0" \
-	"mmcsetup=mmc part\0" \
-	"fdtsetup=fdt addr ${fdtcontroladdr}\0" \
-	"fatenv=fatload mmc ${testpart} ${loadaddr} ${testenv};" \
-		"env import -t ${loadaddr} ${filesize}\0"
-
 #define CONFIG_EXTRA_ENV_SETTINGS \
-	STARLIGHT_FEDORA_BOOTENV \
 	"loadaddr=0xa0000000\0" \
-	STARLIGHT_TEST_BOOTENV \
-	"loadbootenv=fatload mmc ${mmcdev} ${loadaddr} ${bootenv}\0" \
-	"ext4bootenv=ext4load mmc ${bootpart} ${loadaddr} ${bootdir}/${bootenv}\0" \
-	"importbootenv=echo Importing environment from mmc${mmcdev} ...; " \
-		"env import -t ${loadaddr} ${filesize}\0" \
-	"mmcbootenv=setenv bootpart ${mmcdev}:${mmcpart}; " \
-		"mmc dev ${mmcdev}; " \
-		"if mmc rescan; then " \
-			"run loadbootenv && run importbootenv; " \
-			"run ext4bootenv && run importbootenv; " \
-			"if test -n $uenvcmd; then " \
-				"echo Running uenvcmd ...; " \
-				"run uenvcmd; " \
-			"fi; " \
-		"fi\0" \
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE "\0" \
-	BOOTENV \
-	BOOTENV_SF
+	BOOTENV
 
 #define CONFIG_SYS_MAX_FLASH_SECT	0
 #define CONFIG_SYS_MAX_FLASH_BANKS	0
-- 
2.35.1

